### PR TITLE
Reusable media

### PR DESCRIPTION
**Pull request by raphaelm**
- [x] Data model
- [x] Organizer settings
- [x] Item config
- [x] Managing media
  - [x] CRUD
	  - [x] Tests
  - [x] relation to tickets/giftcards
- [x] Media API
  - [x] CRUD
  - [x] Attaching tickets/giftcards
  - [x] Docs
  - [x] Tests
  - [x] device profiles
- [x] order creation API
  - [x] Docs
  - [x] Tests
- [x] Check-in API
	- [x] Tests
	- [x] Docs
	- [x] store used medium with checkin?
- [x] POS support
	- [x] Barcode ticket use case
		- [x] new
		- [x] reuse
		- [x] reuse attendee information
		- [x] print logic
	- [x] nfc gift card use case
	- [ ] merge
- [ ] SCAN support for barcode media
	- [ ] android https://github.com/pretix/libpretixsync/pull/22 https://github.com/pretix/pretixscan-android/pull/62
	- [ ] desktop
	- [ ] ios
	- [ ] proxy

### CODE REVIEW

File: src/pretix/control/templates/pretixcontrol/item/index.html [hunk 1]
```diff
@@ -178,6 +179,12 @@
                     <fieldset>
                         <legend>{% trans "Tickets & Badges" %}</legend>
                         {% bootstrap_field form.generate_tickets layout="control" %}
+                        {% if "media_policy" in form %}
```
Comment (pc-coholic) [inline src/pretix/control/templates/pretixcontrol/item/index.html:182]:
```suggestion
                        {% if form.media_policy %}
```

File: src/pretix/control/templates/pretixcontrol/item/index.html [hunk 2]
```diff
@@ -178,6 +179,12 @@
...
+                            {% bootstrap_field form.media_policy layout="control" %}
+                        {% endif %}
+                        {% if "media_type" in form %}
```
Comment (pc-coholic) [inline src/pretix/control/templates/pretixcontrol/item/index.html:185]:
```suggestion
                        {% if form.media_type %}
```

**Review by pc-coholic — COMMENTED** [review]
- When selecting a `media_type`, also disabled ones are being suggested (Item form and Media Admin view)

File: src/pretix/control/forms/organizer.py [hunk 3]
```diff
@@ -626,6 +663,113 @@ class Meta:
...
+class ReusableMediumCreateForm(ReusableMediumUpdateForm):
+
+    class Meta:
+        model = ReusableMedium
+        fields = ['active', 'type', 'identifier', 'expires', 'linked_orderposition', 'linked_giftcard', 'customer']
+        field_classes = {
+            'expires': SplitDateTimeField,
+            'customer': SafeModelChoiceField,
+            'linked_giftcard': SafeModelChoiceField,
+            'linked_orderposition': SafeOrderPositionChoiceField,
```
Comment (pc-coholic) [inline src/pretix/control/forms/organizer.py:766]: In the EditView, this should not just display the currently selected OrderPosition as `#1 - Ticketname` but rather like in the Select2 Typeahead preview (at least add the Order Code)

**Comment by Arunailae** [comment]
Boss can u teach how to redeem cards

**Comment by pc-coholic** [comment]
I am not 100% happy with `fa-key` as the symbol of reusable media - but there is probably nothing better out there right now in our FA-package... :-/

**Pull request merged**